### PR TITLE
Implement audio backend configuration option

### PR DESCRIPTION
**Pull request by Xpl0itR**
This PR allows the user to select which audio backend to use. If the selected backend is unavailable, then the dummy backend is selected. OpenAL is selected by default.

### CODE REVIEW

**Comment by Xpl0itR** [comment]
Fixed

**Comment by Thog** [comment]
Need rebase

File: Ryujinx/Ui/MainWindow.cs [hunk 1]
```diff
@@ -613,24 +613,24 @@ private static IRenderer InitializeRenderer()
...
         {
-            if (OpenALAudioOut.IsSupported)
+            if (ConfigurationState.Instance.System.AudioBackend.Value == AudioBackend.SoundIo)
             {
-                return new OpenALAudioOut();
-            }
-            else if (SoundIoAudioOut.IsSupported)
-            {
-                return new SoundIoAudioOut();
+                if (SoundIoAudioOut.IsSupported)
```
Comment (AcK77) [inline Ryujinx/Ui/MainWindow.cs:620]: Maybe you could add brackets over it? Same for the "else" and same for the OpenAL condition?

File: Ryujinx/Ui/SettingsWindow.cs [hunk 2]
```diff
@@ -394,14 +404,14 @@ private void SaveToggle_Activated(object sender, EventArgs args)
...
             ConfigurationState.Instance.System.Region.Value                    = Enum.Parse<Configuration.System.Region>(_systemRegionSelect.ActiveId);
+            ConfigurationState.Instance.System.AudioBackend.Value              = Enum.Parse<AudioBackend>(_audioBackendSelect.ActiveId);
             ConfigurationState.Instance.System.TimeZone.Value                  = _systemTimeZoneSelect.ActiveId;
             ConfigurationState.Instance.System.SystemTimeOffset.Value          = _systemTimeOffset;
             ConfigurationState.Instance.Ui.CustomThemePath.Value               = _custThemePath.Buffer.Text;
             ConfigurationState.Instance.Graphics.ShadersDumpPath.Value         = _graphicsShadersDumpPath.Buffer.Text;
             ConfigurationState.Instance.Ui.GameDirs.Value                      = gameDirs;
             ConfigurationState.Instance.System.FsGlobalAccessLogMode.Value     = (int)_fsLogSpinAdjustment.Value;
             ConfigurationState.Instance.Graphics.MaxAnisotropy.Value           = float.Parse(_anisotropy.ActiveId);
-
```
Comment (AcK77) [inline Ryujinx/Ui/SettingsWindow.cs:404]: You shouldn’t remove this empty line

**Comment by Xpl0itR** [comment]
Rebased and addressed comments

**Comment by EmulationFanatic** [comment]
Can we have openal named "OpenAL" (with the properly capitalized L)? Otherwise it looks like openAI (as in artificial intelligence). This is in line with OpenAL's own literature.
image

**Comment by Xpl0itR** [comment]
I've capitalised the names in the GUI

**Pull request merged**